xchange-stream-service-netty/src/main/java/info/bitrich/xchangestream/service/netty/NettyStreamingService.java

```diff
@@ -459,18 +475,24 @@ private void onIdle(ChannelHandlerContext ctx) {
   }
 
   protected void handleChannelMessage(String channel, T message) {
-    NettyStreamingService<T>.Subscription subscription = channels.get(channel);
-    if (subscription == null) {
-      LOG.debug("Channel has been closed {}.", channel);
-      return;
-    }
-    ObservableEmitter<T> emitter = subscription.emitter;
-    if (emitter == null) {
-      LOG.debug("No subscriber for channel {}.", channel);
-      return;
+    if ("ALL".equals(channel)) {
+      channels
+          .forEach((k, v) ->
+              v.emitter.onNext(message));
+
+    } else {
+      NettyStreamingService<T>.Subscription subscription = channels.get(channel);
```

Comment:
should we have NPE protection here given `ConcurrentHashMap` throws them on null keys and same on `handleChannelError` and `subscribeChannel`

Reply:
NettyStreamingService handleMessage/handleError already check before passing to handleChannelMessage/handleChannelError
with !StringUtil.isNullOrEmpty(channel)